File: tournaments/digitsProduct/digitsProduct.js
function digitsProduct(product) {
    let result = 0;
    let store = [];

    if (!product) return 10;
    if (1 === product) return 1;

    for (let divisor = 9; 1 < divisor; --divisor)
        while (!(product % divisor)) {
            product /= divisor;
            store.push(divisor);
        }

    if (1 !== product) return -1;

    for (let i = store.length - 1; 0 <= i; --i)
        result = 10 * result + store[i];

    return result;
}

File: tournaments/digitsProduct/digitsProduct.cpp
int digitsProduct(int product) {

  std::vector<int> answerDigits;
  int answer = 0;

  if (!product) {
    return 10;
  }

  if (product == 1) {
    return 1;
  }

  for (int divisor = 9; divisor > 1; divisor--) {
    while (product % divisor == 0) {
      product /= divisor;
      answerDigits.push_back(divisor);
    }
  }

  if (product > 1) {
    return -1;
  }


  for (int i = (int)answerDigits.size() - 1; i >= 0; i--) {
    answer = 10 * answer + answerDigits[i];
  }
  return answer;
}
